server/commands.go

```diff
@@ -0,0 +1,472 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+	"github.com/pkg/errors"
+)
+
+const (
+	commandIcebreaker                  = "icebreaker"
+	commandIcebreakerAdd               = "icebreaker add"
+	commandIcebreakerApprove           = "icebreaker approve"
+	commandIcebreakerReject            = "icebreaker reject"
+	commandIcebreakerRemove            = "icebreaker remove"
+	commandIcebreakerClearAllProposals = "icebreaker clearall proposals"
+	commandIcebreakerClearAllApproved  = "icebreaker clearall approved"
+	commandIcebreakerShowProposals     = "icebreaker show proposals"
+	commandIcebreakerShowApproved      = "icebreaker show approved"
+	commandIcebreakerResetToDefault    = "icebreaker reset questions"
+)
+
+func (p *Plugin) registerCommands() error {
+	if err := p.API.RegisterCommand(&model.Command{
```

Comment:
This logic can ver very easily de-duplicated.

Instead of registering each command individually, you can create an array of `model.Command` objects, iterate it and register it in the loop.